trading_calendars/exchange_calendar_xjse.py

```diff
@@ -0,0 +1,139 @@
+#
+# Copyright 2019 Quantopian, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from datetime import time
+from pandas.tseries.offsets import Day
+from pandas import Timestamp
+from pandas.tseries.holiday import (
+    Easter,
+    GoodFriday,
+    Holiday,
+    sunday_to_monday,
+)
+from pytz import timezone, UTC
+
+from .common_holidays import new_years_day
+from .trading_calendar import HolidayCalendar, TradingCalendar
+
+
+class XJSEExchangeCalendar(TradingCalendar):
+    """
+    Exchange calendar for the Johannesburg Stock Exchange (XJSE).
+    """
+    name = 'XJSE'
+
+    tz = timezone('Africa/Johannesburg')
+
+    open_times = (
+        (None, time(9, 1)),
+    )
+
+    close_times = (
+        (None, time(17, 00)),
+    )
+
+    @property
+    def regular_holidays(self):
+        return HolidayCalendar([
```

Comment:
We've been defining holidays as variables outside of the class definition, rather than in `regular_holidays()`.  I'm sure you saw this, so maybe you have a reason for defining them here.  I don't see anything wrong with defining them here, but maybe for uniformity it would be good to use the same structure as the rest of the calendars?  Just a thought.

Reply:
I generally prefer to avoid additional name assignments when the object being constructed describes itself with a string literal, but it does look a little cluttered this way. I'll give it a shot.